zcash_client_sqlite/src/wallet/sapling.rs

```diff
@@ -323,36 +324,25 @@ pub(crate) fn select_spendable_sapling_notes<P: consensus::Parameters>(
 /// - No transaction in which the note's nullifier appears has been observed as mined.
 pub(crate) fn get_sapling_nullifiers(
     conn: &Connection,
+    query: NullifierQuery,
 ) -> Result<Vec<(AccountId, Nullifier)>, SqliteClientError> {
     // Get the nullifiers for the notes we are tracking
-    let mut stmt_fetch_nullifiers = conn.prepare(
-        "SELECT rn.id, rn.account_id, rn.nf, tx.block as block
-         FROM sapling_received_notes rn
-         LEFT OUTER JOIN transactions tx
-         ON tx.id_tx = rn.spent
-         WHERE block IS NULL
-         AND nf IS NOT NULL",
-    )?;
-    let nullifiers = stmt_fetch_nullifiers.query_and_then([], |row| {
-        let account = AccountId(row.get(1)?);
-        let nf_bytes: Vec<u8> = row.get(2)?;
-        Ok::<_, rusqlite::Error>((account, sapling::Nullifier::from_slice(&nf_bytes).unwrap()))
-    })?;
-
-    let res: Vec<_> = nullifiers.collect::<Result<_, _>>()?;
-    Ok(res)
-}
+    let mut stmt_fetch_nullifiers = match query {
+        NullifierQuery::Unspent => conn.prepare(
+            "SELECT rn.id, rn.account_id, rn.nf
```

Comment:
Do we need to select `rn.id` anymore? IIRC that was because we used to pass around the note ID in an opaque struct, but we don't do that anymore. Is the ID select serving any remaining purpose?

Reply:
Looks like not; we don't query that index in the returned row.